source/common/secret/secret_manager_impl.cc

```diff
@@ -47,17 +57,8 @@ TlsCertificateConfigProviderSharedPtr SecretManagerImpl::findOrCreateDynamicSecr
   if (!secret_provider) {
     ASSERT(secret_provider_context.initManager() != nullptr);
 
-    std::function<void()> unregister_secret_provider = [map_key, config_name, sds_config_source,
-                                                        this]() {
-      ENVOY_LOG(debug, "Unregister secret provider. name: {}, sds config: {}", config_name,
-                sds_config_source.DebugString());
-      auto secret_provider = dynamic_secret_providers_.find(map_key);
-      if (secret_provider != dynamic_secret_providers_.end()) {
-        dynamic_secret_providers_.erase(map_key);
-      } else {
-        ENVOY_LOG(error, "secret provider does not exist. name: {}, sds config: {}", config_name,
-                  sds_config_source.DebugString());
-      }
+    std::function<void()> unregister_secret_provider = [map_key, this]() {
+      this->removeDynamicSecretProvider(map_key);
```

Comment:
may not need this variable, just use the lambda directly in the function argument. such

xxx(....,
[map_key, this] () { removeDynamicSecretProvider(map_key); } 
);